### PR TITLE
OADP-4135: rebase upstream release-1.10 changes for volume_snapshotter

### PR DESCRIPTION
**Pull request by kaovilai**
Take current oadp-1.4, and merge the changes against velero release-1.10's volume_snapshotter.go keeping "wait for snapshot ready" features.

Signed-off-by: Tiger Kaovilai <tkaovila@redhat.com>

### CODE REVIEW

File: velero-plugin-for-aws/volume_snapshotter.go [hunk 1]
```diff
@@ -60,120 +62,106 @@ func newVolumeSnapshotter(logger logrus.FieldLogger) *VolumeSnapshotter {
...
-	// if config["snapshotCreationTimeout"] is empty, default to 60m; otherwise, parse it
-	var err error
-	if val := config[snapshotCreationTimeoutKey]; val == "" {
-		b.snapshotCreationTimeout = snapshotCreationTimeoutDefault
-	} else {
-		b.snapshotCreationTimeout, err = time.ParseDuration(val)
-		if err != nil {
-			return errors.Wrapf(err, "unable to parse value %q for config key %q (expected a duration string)", val, snapshotCreationTimeoutKey)
-		}
-	}
```
Comment (kaovilai) [inline velero-plugin-for-aws/volume_snapshotter.go]: missed this init timeout.

**Comment by weshayutin** [comment]
make test and make container passed for me.  Thank you @kaovilai !   @sseago please review

File: velero-plugin-for-aws/config.go [hunk 2]
```diff
@@ -4,14 +4,15 @@ import (
 	"context"
 	"crypto/tls"
 	"crypto/x509"
+	"net/http"
+	"os"
+
 	"github.com/aws/aws-sdk-go-v2/aws"
```
Comment (sseago) [inline velero-plugin-for-aws/config.go]: These changes aren't related or needed to fix the commit, so we should leave this as it is from upstream -- just remove the file from the PR.

File: velero-plugin-for-aws/volume_snapshotter.go [hunk 3]
```diff
@@ -17,15 +17,17 @@ limitations under the License.
...
 	"strings"
 	"time"
 
-	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/aws/awserr"
-	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/aws/smithy-go"
+
```
Comment (sseago) [inline velero-plugin-for-aws/volume_snapshotter.go:30]: Looking at a diff from this PR to upstream release-1.10, these imports are added in a different place, so the resulting PR will be moving imports. Best to put them in the same place here as they are upstream so it doesn't add to future merge conflict points.

**Review by sseago — COMMENTED** [review]
Other than the issues noted with the imports, I think this is just about right.

**Review by sseago — APPROVED** [review]

**Pull request merged**